imgui-knobs.cpp

```diff
@@ -170,7 +195,14 @@ namespace ImGuiKnobs {
 
             // Draw input
             if (!(flags & ImGuiKnobFlags_NoInput)) {
-                auto changed = ImGui::DragScalar("###knob_drag", data_type, p_value, speed, &v_min, &v_max, format);
+                ImGuiSliderFlags drag_scalar_flags = 0;
+                if (!(flags & ImGuiKnobFlags_NoAlwaysClamp)) {
```

Comment:
I read your comment about the AlwaysClamp, but I tested with the NoAlwaysClamp and zero values and couldn't immediately see anything wrong happening? Can you give an example of when the default behaviour is unexpected? I'd like to keep the ImGuiKnobFlags to a minimum.

Reply:
I need ImGuiSliderFlags_AlwaysClamp in ImGui::DragScalar to avoid my users to enter zero or negative frequencies. 

I don't want to add extra code for clamping in the client code for this. Also the other ImGui Widgets like drags and sliders support this flag. Your knob widgets currently do not support this flag https://github.com/altschuler/imgui-knobs/blob/aa18aefa0a21a9a9c2d67dd0b6514a89e1da9230/imgui-knobs.cpp#L173 

If you want I can remove the extra flag, but I would prefer to keep this option for flexibility.

Options:
1) leave it as is is (ImGuiKnobFlags_NoAlwaysClamp)
2) invert it to be consistent with the upstream ImGui Widgets? So instead of ImGuiKnobFlags_NoAlwaysClamp name it ImGuiKnobFlags_AlwaysClamp (like ImGuiSliderFlags_AlwaysClamp)
3) provide an additional parameter to pass through ImGuiSliderFlags
4) remove the flag but keep the ImGuiSliderFlags_AlwaysClamp behavior internally as hard coded default.

Reply:
Ah that makes good sense. I think option 2 is the cleanest. Option 3 would make sense, but I'd like to keep the API minimal, and not all the flags are necessarily useful/relevant to sliders, so let's stick with mirroring what we need.

Reply:
fixed in f108c4e1ed084eb002fd19d1e851d4565400ac8e